core/dbt/task/compile.py

```diff
@@ -40,6 +40,10 @@ def compile(self, manifest):
 
 
 class CompileTask(GraphRunnableTask):
+    # We add a new inline node to the manifest during initialization
+    # it should be removed before the task is complete
+    _inline_node_id = None
```

Comment:
It feels a bit off to have a non-constant, instance-specific representation state as a class attribute. Is there a reason this isn't defined on the instance instead via an __init__ override?